contracts/zero-ex/contracts/src/features/BundleSwapFeature.sol

```diff
@@ -80,11 +80,15 @@ contract BundleSwapFeature is
                 }
             } else {
                 // cannot ErrorBehaviour.STOP or CONTINUE - always has to revert to not lose sent ETH
+                // REVIEW: I think the issue here might occur even for erc20 cases, the swap can fail
+                // after transfers have been made, so maybe we need to check that input amount of token
+                // is still held the owner, which can be tricky...
                 if (LibERC20Transformer.isTokenETH(swap.inputToken)) {
                     revert("BundleSwapFeature::bundleSwap/SELL_ETH_REVERTED");
                 }
 
                 // also revert if first trade
+                // REVIEW: I think here the check meant is `i == 0` instead of `length == 0`
```

Comment:
ah.. it should be length == 1 instead of 0
good spot!

The reason of this check is to revert in case there is just one swap to safe gas fees for everything that happens after this.

Reply:
adapted code and comment